src/v/iceberg/manifest_file_packer.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_file_packer.h"
+
+namespace iceberg {
+
+namespace {
+struct bin {
+    explicit bin(size_t target)
+      : target_weight(target)
+      , total_weight(0) {}
+    const size_t target_weight;
+    size_t total_weight;
+    chunked_vector<manifest_file> items;
+
+    bool can_add(const manifest_file& f) const {
+        if (total_weight == 0) {
+            return true;
+        }
+        return total_weight + f.manifest_length <= target_weight;
+    }
+
+    void add(manifest_file f) {
+        total_weight += f.manifest_length;
+        items.emplace_back(std::move(f));
+    }
+
+    // Resets this bin, returning the items.
+    chunked_vector<manifest_file> reset() {
+        total_weight = 0;
+        return std::move(items);
```

Comment:
nit: this should really be `return std::exchange(items, {})` to express that items container is reset.

Reply:
Done